components/earnings/commons/earnings_base.lua

```diff
@@ -153,9 +151,9 @@ function Earnings.calculate(conditions, queryYear, mode, perYear, aliases, isPla
 	local sumUp = function(placement)
 		local value = Earnings._determineValue(placement, aliases, isPlayerQuery)
 		if perYear then
+			---@type integer?
```

Comment:
This one can probably be removed, as `tonumber` will imply `number?` (and in lua 5.1 there's no difference between integer and number).